osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/AbstractEncryptionProvider.java

```diff
@@ -28,28 +27,28 @@
 
     public abstract EncryptionProviderType getType();
 
-    protected abstract Cipher getCipher();
+    protected abstract Cipher getCipher() throws EncrypterException;
 
     protected abstract AlgorithmParameterSpec getAlgorithmParameterSpec();
 
-    protected abstract Key getSecretEncryptionKey(String keyReference, int cipherMode);
+    protected abstract Key getSecretEncryptionKey(String keyReference, int cipherMode) throws EncrypterException;
 
-    protected void setKeyFile(File keyFile) {
+    protected void setKeyFile(final File keyFile) {
         this.keyFile = keyFile;
     }
 
-    public EncryptedSecret encrypt(Secret secret, String keyReference) {
+    public EncryptedSecret encrypt(final byte[] secret, final String keyReference) throws EncrypterException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.EncrypterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)